spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -222,11 +222,11 @@ public SparkTable alterTable(Identifier ident, TableChange... changes) throws No
     try {
       Table table = load(ident);
       commitChanges(table, setLocation, setSnapshotId, pickSnapshotId, propertyChanges, schemaChanges);
+      table.refresh();
+      return new SparkTable(table, false);
```

Comment:
Please add a comment to explain the boolean argument here.

Reply:
I set  `refreshEagerly` here to `false` because I think it makes sense to returns a table with the state right after the update. Otherwise, if I understand correctly, the table might be updated from other places unintentionally. And this will cause bugs on code that relies on the table state right after the update.

Besides, it seems that we always set this to false throughout the codebase atm.

- `cacheEabled` is `true` by default: https://github.com/apache/iceberg/blob/36e261bb40a0502185621165e0fa3441d4273d11/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L122
- https://github.com/apache/iceberg/blob/36e261bb40a0502185621165e0fa3441d4273d11/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java#L140
- https://github.com/apache/iceberg/blob/36e261bb40a0502185621165e0fa3441d4273d11/spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java#L616

Reply:
I can still add an inline comment if you think that's needed

Reply:
Yep, please do. We can follow up with a PR to fix the other case that you found passing `false`.